### PR TITLE
Javascript2 week1 is ready to comments

### PR DESCRIPTION
**Pull request by aijazraja**

### CODE REVIEW

File: javascript/javascript2/week1/index.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
+             /********************************************************/
+               console.log('\n        Smallest letter in Array\n\n')
+             /********************************************************/      
+
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+const smalleLetter=getSmallestLetter(danishWords); // returns 'ø'
+function getSmallestLetter(words){
+    let temp=danishWords[0];
+    for(let i=0;i<danishWords.length;i++){
```
Comment (Rassibassi) [inline javascript/javascript2/week1/index.js]: In the first loop iteration your code will always compare the first word with itself!
E.g., here in the first iteration both word and shortestWord are "bil"
Which means there is always one unnecessary iteration. Can you change the for loop such that this is solved?

File: javascript/javascript2/week1/index.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+             /********************************************************/
+               console.log('\n        Smallest letter in Array\n\n')
+             /********************************************************/      
+
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+const smalleLetter=getSmallestLetter(danishWords); // returns 'ø'
+function getSmallestLetter(words){
+    let temp=danishWords[0];
```
Comment (Rassibassi) [inline javascript/javascript2/week1/index.js]: `temp`? bad variable name

File: javascript/javascript2/week1/index.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    for(let i=0;i<danishWords.length;i++){
+         if(danishWords[i].length<=temp.length){
+             temp=danishWords[i];
+             danishWords[i]=danishWords[0];
+             danishWords[0]=temp;
+
```
Comment (Rassibassi) [inline javascript/javascript2/week1/index.js:14]: why are you assigning the variable so weirdly here? Dont you only need `temp = danishWords[i]` and later `return temp`

File: javascript/javascript2/week1/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+document.getElementById('smallestLetter').innerHTML='Smallest letter in array : '+smalleLetter;
+console.log('Smallest letter in array ["bil", "plante", "kaffe", "bog", "ø", "planetarium"] is : '+smalleLetter);
+
+
+                /********************************************************/
+                      console.log('\n        Danish letters count\n\n')
+                /********************************************************/ 
+
+const danishString2 = "Blå grød med røde bær";
+let danishLetters=danishLettersCounter(danishString2); // returns {total: 4, æ: 1, ø: 2, å: 1}
```
Comment (Rassibassi) [inline javascript/javascript2/week1/index.js:31]: you function does not return `{total: 4, æ: 1, ø: 2, å: 1}` but a string. Also if a letter has the zero count, you list it, where the solution should not be listing letters with zero count

File: javascript/javascript2/week1/index1.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    <div>
+        <h1>Spirit animal name generator</h1>
+        <input type="text" id="name"> <br>
+        <br><button id="getname" type="submit">Get full Name</button><br>
+        <h1 id="h1"></h1> <br>
+
+    </div>
+    <div>
+        <h1>Upgrated spirit animal name generator</h1>
+        <select name="" id="select" onchange="onChange()">
```
Comment (Rassibassi) [inline javascript/javascript2/week1/index1.html]: Please us `addEventListener` to add the `onchange` event to the select tag. its not good practice to mix up event listeners in html and javascript.

File: javascript/javascript2/week1/main.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+const products = getAvailableProducts();
+//console.log(products.length);
+
+function randerProducts(products){
+    let ulList=document.getElementById("productList");
+
+    for(let i=0;i<products.length;i++){
+    const liElement=document.createElement("li");
+    liElement.innerHTML='<h2>'+products[i].name +'</h2> Price:'+products[i].price+'<br> Rating:'+products[i].rating+'<br> <br>'
```
Comment (Rassibassi) [inline javascript/javascript2/week1/main.js]: This works, but didn't we learn how to create `h2` tags in javascript via the DOM API?

File: javascript/javascript2/week1/index.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+ let firstName=document.getElementById('getname');
+ firstName.addEventListener('click',getSpiritName);
+ function getSpiritName(){
+     let spiritAnimals=['-The Aardwolf','-The Alley Cat','-The Bear','-The Black Jaguar',
+     '-The Dolphin','-The Falcon','-The Gorilla','-The Hawk','-The Honey Bee','-The Jaguar'];
+     let randomAnimal=spiritAnimals[Math.floor(Math.random()*10 )];
+    let name1=document.getElementById('name').value;
+     if(name1==''){
+         let a='Please enter name'
```
Comment (Rassibassi) [inline javascript/javascript2/week1/index.js]: bad variable name, and why not `const`?

**Review by Rassibassi — COMMENTED** [review]
Good work, let me know if you have any questions regarding my comments

File: javascript/javascript2/week1/index.js (hunk 1 above)
Comment (aijazraja) [reply javascript/javascript2/week1/index.js]: done

File: javascript/javascript2/week1/index.js (hunk 2 above)
Comment (aijazraja) [reply javascript/javascript2/week1/index.js]: done

File: javascript/javascript2/week1/index.js (hunk 3 above)
Comment (aijazraja) [reply javascript/javascript2/week1/index.js:14]: done

File: javascript/javascript2/week1/index.js (hunk 4 above)
Comment (aijazraja) [reply javascript/javascript2/week1/index.js:31]: done..returning object now

File: javascript/javascript2/week1/index1.html (hunk 5 above)
Comment (aijazraja) [reply javascript/javascript2/week1/index1.html]: done

File: javascript/javascript2/week1/index.js (hunk 7 above)
Comment (aijazraja) [reply javascript/javascript2/week1/index.js]: fixed

**Pull request merged**